### DISCUSSION TITLE
Disable LSP based on filetype

### DISCUSSION THREAD

**Discussion by idelice** [LazyVim/LazyVim#2281] [General]
I have both yamlls and helm_ls LSP's installed and attached when entering any yaml file + helm chart.

I also have the extras yaml file lazy loaded and a helm plugin.

```lua
return {
  "towolf/vim-helm",
  dependencies = {
    "neovim/nvim-lspconfig",
  },
  config = function()
    local configs = require("lspconfig.configs")
    local lspconfig = require("lspconfig")
    local util = require("lspconfig.util")

    if not configs.helm_ls then
      configs.helm_ls = {
        default_config = {
          cmd = { "helm_ls", "serve" },
          filetypes = { "helm" },
          root_dir = function(fname)
            return util.root_pattern("Chart.yaml")(fname)
          end,
        },
      }
    end

    lspconfig.helm_ls.setup({
      filetypes = { "helm" },
      cmd = { "helm_ls", "serve" },
    })
  end,
}
```

I want to disable yamlls LSP when the filetype is 'helm' as the LSP's are conflicting.

How do I approach this?

**Comment by dpetka2001** [comment]
See this [answer](https://github.com/LazyVim/LazyVim/discussions/1389#discussioncomment-6893131) and make the necessary changes for your case. It should be similar.

**Comment by idelice** [comment]
> See this [answer](https://github.com/LazyVim/LazyVim/discussions/1389#discussioncomment-6893131) and make the necessary changes for your case. It should be similar.

That worked thank you.

I made the following changes in case anyone stumbles upon the same thing:

```lua
require("lazyvim.util").lsp.on_attach(function(client, buffer)
  if client.name == "yamlls" then
    if vim.api.nvim_buf_get_option(buffer, "filetype") == "helm" then
      vim.schedule(function()
        vim.cmd("LspStop ++force yamlls")
      end)
    end
  end
end)

return {
  "towolf/vim-helm",
  ft = "helm",
  lazy = true,
  dependencies = {
    "neovim/nvim-lspconfig",
    opts = {
      servers = {
        helm_ls = {},
        yamlls = {},
      },
    },
  },
}
```

**Reply by dpetka2001** [reply]
You don't need `lazy=true`. Since you have `ft="helm"`, it implicitly means it will lazy-load on the `ft` you specified.

**Reply by idelice** [reply]
Fixed that

**Comment by idelice** [comment]
@dpetka2001 I'll have to reopen this discussion because I found an issue with the current approach:

If I start by opening a file with the filetype of 'helm', i can confirm that `helm_ls` LSP is the only one attached to the buffer, but if I open a normal yaml file next, I can see  `helm_ls` LSP is attached - which it **shouldn't**. 

But it doesn't end there! If i get back to my helm file, I can see that `yamlls` LSP is once attached..

How do I approach this?

**Reply by dpetka2001** [reply]
I don't use it myself, so I believe if you ping the user who gave the answer in the other thread, it might be more helpful. Can you link somewhere an exampe of helm file that I could use for testing?

**Reply by idelice** [reply]
I think I may have looked wrong at it.. even tho it shows client `yamlls`, it also says that it is _not_ attached to the current buffer. This is enough for me :) thank you
